src/eventsSince.ts

```diff
@@ -0,0 +1,85 @@
+'use strict'
+// chel eventsSince [--limit N] <url-or-localpath> <contractID> <hash>
+
+import { flags, path } from './deps.ts'
+import { exit, isDir, isFile } from './utils.ts'
+
+const headPrefix = 'head='
+
+function isURL (arg: string): boolean {
+  return URL.canParse(arg) && Boolean(new URL(arg).host)
+}
+
+const backends = {
+  fs: await import('./database-fs.ts'),
+  sqlite: await import('./database-sqlite.ts')
+}
+
+let backendFrom: typeof backends.sqlite | typeof backends.fs
+
+export async function eventsSince (args: string[]): Promise<string[]> {
+  const parsedArgs = flags.parse(args)
+
+  const limit = Number.parseInt(parsedArgs.limit ?? 50)
+  const [urlOrLocalPath, contractID, hash] = parsedArgs._
+  const src = urlOrLocalPath
+
+  let from
+  if (isDir(src)) {
+    from = 'fs'
+  } else if (isFile(src)) {
+    from = 'sqlite'
+  } else if (isURL(src)) {
+    from = 'remote'
+  } else {
+    exit(`invalid argument: "${src}"`)
+  }
+
+  backendFrom = backends[from as keyof typeof backends]
+  try {
+    await backendFrom.initStorage(from === 'fs' ? {dirname: src} : {dirname: path.dirname(src), filename: path.basename(src)})
+  } catch (error) {
+    exit(`could not init storage backend at "${src}" to fetch events from: ${error.message}`)
+  }
+  const messages = await getMessagesSince(contractID, hash, limit)
+  console.log(Deno.inspect(messages.map(s => JSON.parse(s)), { colors: true, strAbbreviateSize: Infinity }))
```

Comment:
This looks great by itself, but it isn't parseable by the `jq` command because the output is not JSON. E.g. it displays objects and their keys are not wrapped in quotes.

We need to be able to run commands like this:

```sh
chel eventsSince groupincome.db 21XWnNJZQoeFQsmrSQm4FNKSVbErWHDsszBemXRgS1ZWPurSFs 21XWnNJZQoeFQsmrSQm4FNKSVbErWHDsszBemXRgS1ZWPurSFs | jq '.[] | select(.op[0] == "ae") | .op[1] | fromjson'
```

In order to do that this line needs to be:

```js
console.log(JSON.stringify(messages.map(s => JSON.parse(s)), null, 2))
```

Reply:
Ok 👍